src/main/java/de/tum/cit/aet/artemis/core/web/CalendarResource.java

```diff
@@ -0,0 +1,120 @@
+import de.tum.cit.aet.artemis.core.util.CalendarUtil;
+import de.tum.cit.aet.artemis.exam.api.ExamApi;
+import de.tum.cit.aet.artemis.exercise.service.ExerciseService;
+import de.tum.cit.aet.artemis.lecture.api.LectureApi;
+import de.tum.cit.aet.artemis.quiz.service.QuizExerciseService;
+import de.tum.cit.aet.artemis.tutorialgroup.api.TutorialGroupApi;
+
+@Lazy
+@Profile(PROFILE_CORE)
+@RestController
+@RequestMapping("api/core/")
+public class CalendarResource {
+
+    private static final Logger log = LoggerFactory.getLogger(CalendarResource.class);
+
+    private final UserRepository userRepository;
+
+    private final TutorialGroupApi tutorialGroupApi;
+
+    private final LectureApi lectureApi;
+
+    private final ExamApi examApi;
+
+    private final QuizExerciseService quizExerciseService;
+
+    private final ExerciseService exerciseService;
+
+    private final CourseRepository courseRepository;
+
+    private final AuthorizationCheckService authorizationCheckService;
+
+    public CalendarResource(UserRepository userRepository, TutorialGroupApi tutorialGroupApi, LectureApi lectureApi, ExamApi examApi, ExerciseService exerciseService,
+            QuizExerciseService quizExerciseService, CourseRepository courseRepository, AuthorizationCheckService authorizationCheckService) {
+        this.userRepository = userRepository;
+        this.tutorialGroupApi = tutorialGroupApi;
+        this.lectureApi = lectureApi;
+        this.examApi = examApi;
+        this.quizExerciseService = quizExerciseService;
+        this.exerciseService = exerciseService;
+        this.courseRepository = courseRepository;
+        this.authorizationCheckService = authorizationCheckService;
+    }
+
+    /**
+     * GET /course/:courseId/calendar-events : gets all {@link CalendarEventDTO}s associated to the given course falling into the requested month
+     *
+     * @param courseId  the id of the course for which the events should be fetched
+     * @param monthKeys a list of ISO 8601 formatted strings representing months
+     * @param timeZone  the clients time zone as IANA time zone ID
+     * @return {@code 200 (OK)} with a map of DTOs keyed by day from client timezone perspective. All timestamps conform to ISO 8601 format.
+     * @throws EntityNotFoundException  {@code 404 (Not Found)} if no course exists for the provided courseId
+     * @throws AccessForbiddenException {@code 403 (Forbidden)} if the user does not have at least student role or if the user is not at least student in the course
+     * @throws BadRequestException      {@code 400 (Bad Request)} if the monthKeys are empty or formatted incorrectly or if the timeZone is formatted incorrectly.
+     */
+    @GetMapping("calendar/courses/{courseId}/calendar-events")
+    @EnforceAtLeastStudent
+    public ResponseEntity<Map<String, List<CalendarEventDTO>>> getCalendarEventsOverlappingMonths(@PathVariable Long courseId, @RequestParam List<String> monthKeys,
```

Comment:
The courseId should probably not be nullable here.
```suggestion
    public ResponseEntity<Map<String, List<CalendarEventDTO>>> getCalendarEventsOverlappingMonths(@PathVariable long courseId, @RequestParam List<String> monthKeys,
```